### ISSUE TITLE
[FEATURE] Allow super-admin to control which system indices are not permission-able

### ISSUE THREAD

**Issue by DarshitChanpura** [opensearch-project/security#3038]
Related to https://github.com/opensearch-project/security/issues/2553

With #2887, a user who has access to modify roles will now have the ability to grant other user with permissions to the system indices. There is a potential of misuse here. If a rogue user, gets access to `security` index they can delete it causing no users to be able to access the cluster other than the `super-admin` who will have to create the index again.

**What solution would you like?**

Create a "denylist" setting which contains a list of system indices that are not permission-able to normal users.

**Comment by scrawfor99** [comment]
[Triage] Hi @DarshitChanpura, thank you for filing this issue. I am going to leave this issue untriaged pending further review of whether treating the security index separately is adequate.

**Comment by DarshitChanpura** [comment]
[Update from Security review] Having security index isolated and documenting it should suffice in satisfying the requirement for this issue.

**Comment by samuelcostae** [comment]
@DarshitChanpura  _"list of system indices that are not permission-able to normal user"_

Only super admin would be able to access it by bypassing the checks, is that correct?

**Comment by scrawfor99** [comment]
[Triage] @DarshitChanpura per your last message "Having security index isolated and documenting it should suffice in satisfying the requirement for this issue." Is there an issue addressing this specific concern? If so we can close this. If there is not, would you want to convert this issue accordingly.

**Comment by scrawfor99** [comment]
[Triage] @DarshitChanpura following up

**Comment by peternied** [comment]
This is duplicate of https://github.com/opensearch-project/security/issues/2845

**Comment by DarshitChanpura** [comment]
#2845 talks about adding an explicit feature flag that enables/disables the feature. This issue talks about maintaining an explicit deny-list of system indices which are not permission-able (this list will only contain security-index as of now). Let's close it once #2887 is merged.

**Comment by scrawfor99** [comment]
[Triage] This issue is incorporated into the work of #2887, so we can leave this open as an actionable triaged issue until that PR is merged.

**Comment by DarshitChanpura** [comment]
In proposed implementation, only security index is non permission-able and the deny list remains non-configurable ([the check will be hardcoded as part of the PR](https://github.com/opensearch-project/security/pull/2887/files#diff-5fce73d989856a9476f3c40b358f26887f92b05d87696d2c2edd5a6b8e1e4a37R78)). We can file a new issue in feature if there is a requirement for a broader and more dynamically configurable deny list.
